### DISCUSSION TITLE
How can I check whether a language standard is present?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1712] [Q&A]
```
option ("some_feature")
    ...
    after_check(function (option)
        -- if the compiler does not support c++17
        option:enable(false)
    end)
```
可以用判断平台+各平台has_flags实现，但这里有通用的实现方法吗？

**Comment by waruqi** [comment]
目前只能走 has_flags

https://xmake.io/#/zh-cn/manual/extension_modules?id=detecthas_flags

**Reply by xq114** [reply]
https://github.com/xmake-io/xmake/issues/1528 能不能扩展一下has_features接口支持一下standard探测，这样对没有提供feature detection的部分也可以用standard探测来设置？

**Reply by waruqi** [reply]
你可以提个feature request过来先备着
